training/MESSAGING_OVERVIEW.md

```diff
@@ -0,0 +1,138 @@
+# Communication Between Client and Server
+This document presents a high-level overview of Arkouda's Client and Server interaction. By the end of this document you should be able to create a Python module that sends a request to the server, receive this request on the server in your own new Chapel module, and return a message back to the Client.
+
+<a id="toc"></a>
+## Table of Contents
+1. [Overview](#msgOverview)
+2. [Sending a Message to the Server](#sendReq)
+3. [Recieving a Request on the Server](#receiveReq)
+4. [Creating a new Chapel Module](#newModule)
+5. [Try-It Example](#tryIt)
+
+<a id="msgOverview"></a>
+## Overview
+
+Arkouda uses ZeroMQ to handle messaging between our Python client and Chapel Server. This communication enables Arkouda to process large scale python data frameworks efficiently by leveraging Chapel's compiled and parallel nature. 
+
+To accomplish this, we use a command map structure in Chapel to map a string name to a Chapel function within a server Module. Adding a new module is simple. At the root level there is a `ServerModules.cfg` file that lists all Chapel Modules available for Command Map arguments. To add a new module, simply add the name of the module to this file. To learn how to structure a new Chapel Module, go to [Creating a New Module](#newModule).
+
+<a id="sendReq"></a>
+## How to Send a Request to the Server
+
+In general, a server request will look something like this:
+
+`repMsg = cast(str, generic_msg(cmd="serverFunction", args={"arg1": arg1, "arg2": arg2}))`
+
+We cast the return from `generic_msg` to a `str` in most cases to ensure our return is in a string format. The alternative is `memoryview`, which is a binary representation.

```

Comment:
We really should explain when `memoryview` comes into play.  I also think this blob should be after breaking down the parameters of `generic_msg`.